game-app/game-core/src/main/java/games/strategy/engine/framework/HistorySynchronizer.java

```diff
@@ -26,8 +27,16 @@
         public void gameDataChanged(final Change change) {
           SwingUtilities.invokeLater(
               () -> {
-                final Change localizedChange = (Change) translateIntoMyData(change);
-                gameData.getHistory().getHistoryWriter().addChange(localizedChange);
+                try {
+                  final Change localizedChange = (Change) translateIntoMyData(change);
+                  gameData.getHistory().getHistoryWriter().addChange(localizedChange);
+                } catch (ConcurrentModificationException e) {
+                  // Temporary instrumentation to diagnose root cause of:
+                  // https://github.com/triplea-game/triplea/issues/10274
+                  // Include the Change subclass name in the error.
+                  throw new ConcurrentModificationException(
+                      "Translating " + change.getClass().getName(), e);
```

Comment:
The exception message is what would be shown to a user in the error prompt. This exception message won't really tell them what is happening. Ideally all error message state what the code was trying to do, what failed, and what the user should do about the error. In this case, the user should do very little other than "please report this to TripleA, error translating change class: " + change.getClass().getName().

But, I do get this is temporary code and is intended to be removed before release (notably because it would be fixed by then hopefully too)

Reply:
To be fair, without this, there's no message and it's just a "java.util.ConcurrentModificationException".